applications/MIXERv2.md

```diff
@@ -0,0 +1,172 @@
+
+### Team Code Repos
+- https://github.com/webb-tools/arkworks-gadgets
+- https://github.com/webb-tools/anon
+
+### Team LinkedIn Profiles
+- https://www.linkedin.com/in/drewcstone/
+
+## Development Roadmap :nut_and_bolt: 
+### Overview
+* **Total Estimated Duration:** 4 months
+* **Full-time equivalent (FTE):**  5 FTE
+* **Total Costs:** $100,000 DAI
+
+### Milestone 1 — Implement multi-currency, multi-backend, multi-hash support in Substrate modules and Arkworks mixer circuits
+* **Estimated Duration:** 2 months
+* **FTE:** 4
+* **Costs:** $60,000
+
+| Number | Deliverable | Specification |
+| ------------- | ------------ | ------------- |
+| 0a. | License | Apache 2.0 / MIT / Unlicense |
+| 0b. | Documentation | We will provide both inline documentation of the code and a basic tutorial that explains how a user can (for example) spin up one of our Substrate nodes. Once the node is up, it will be possible to send test transactions that will show how the new functionality works. |
+| 0c. | Testing Guide | The code will have unit-test coverage (min. 70%) to ensure functionality and robustness. In the guide we will describe how to run these tests | 
+| 0d. | Article/Tutorial | We will write an article or tutorial that explains the work done as part of the grant. 
+| 1. | Substrate module: Multi-Asset support for mixer | We will extend the mixer for ORML libraries and pallet-assets libraries. |
+| 1a. | Substrate module: ORML support | Plug and play functionality for ORML tokens |
+| 1b. | Substrate module: Pallet asset support | Plug and play functionality for Asset module |
+| 2. | Substrate module: New ZKP backend support | We will add new backends to our pallets to allow users more flexibility over their target ZKP system. |
+| 2a. | Substrate module: New ZKP backend support | Groth16 style proofs using BLS381, BN254 |
+| 2b. | Substrate module: ZKP backend support | Bulletproof style proofs using Curve25519 |
+| 2c. | Substrate module: New ZKP hash function support | More Poseidon configurations, MIMC |
+| 3. | Zero-knowledge circuits | We will create new zero-knowledge circuits for mixer functionality using the Arkworks backend, which allows us to plug and play with different elliptic curves and SNARKs. |  
+| 3. | Docker | We will provide a dockerfile to demonstrate the full functionality of our chain |
+
+# Additional task details
+
+**Pallet:**
+
+- [ ] Create generalised structure for integrating new zero-knowledge backends into the mixer.
+    - [ ]  Add support to the merkle tree pallet for circuit specific setups using arbitrary backends (Groth16, Bulletproofs)
+    - [ ]  Augment mixer data structure with setup/backend.
+    - [ ]  Tests
```

Comment:
zk storage is important.  I'm quite happy with this goal, even ignoring everything else.